src/js/components/DataTable/Body.js

```diff
@@ -181,20 +182,42 @@ const Body = forwardRef(
 
     return (
       <Keyboard
-        onEnter={
-          // active is undefined if user never used the keyboard
-          onClickRow &&
-          active >= 0 &&
-          (!disabled ||
-            !disabled.includes(datumValue(data[active], primaryProperty)))
-            ? (event) => {
-                event.persist();
-                const adjustedEvent = event;
-                adjustedEvent.datum = data[active];
-                onClickRow(adjustedEvent);
-              }
-            : undefined
-        }
+        onEnter={(event) => {
+          if (
+            onClickRow &&
+            active >= 0 &&
+            (!disabled ||
+              !disabled.includes(datumValue(data[active], primaryProperty)))
+          ) {
+            if (typeof onClickRow === 'function') {
+              event.persist();
+              const adjustedEvent = event;
+              adjustedEvent.datum = data[active];
+              onClickRow(adjustedEvent);
+            } else if (onClickRow === 'onSelect') {
+              const primaryValue = data[active]?.[primaryProperty];
+              if (selected && selected.includes(primaryValue)) {
+                onSelect(selected.filter((s) => s !== primaryValue));
+              } else onSelect([...selected, primaryValue]);
+            }
+          }
+        }}
+        // The WCAG recommendation for checkboxes is to select them with "Space"
+        onSpace={() => {
+          if (
```

Comment:
Instead of duplicating this code for `onEnter` and `onSpace`, is there any way to create reusable function that both use?

Reply:
yes I pulled this out but left the `If` statement for the first part